In [1]:
from helpers import parse_and_count_min_max_avg_predicates

schema_org_classes = [
    "Answer",
    "Book",
    "Dataset",
    "Hospital",
    "Movie",
    "MusicAlbum",
    "QAPage",
    "Recipe",
    "School",
    "ShoppingCenter"
]

filenames = {}

filenames['Answer'] = []
for i in range(118):
    filenames['Answer'].append(rf"E:\schema_org_datasets\Answer\Answer_part_{i}.gz")
    
filenames['Book'] = []
for i in range(19):
    filenames['Book'].append(rf"E:\schema_org_datasets\Book\Book_part_{i}.gz")
    
filenames['Dataset'] = []
for i in range(1, 6):
    filenames['Dataset'].append(f"../dataset_2024_p{i}.gz")
    
filenames["Hospital"] = ["../hospital_2024_p1.gz", "../hospital_2024_p2.gz"]

filenames["Movie"] = []
for i in range(1, 13):
    filenames["Movie"].append(f"../movie_2024_p{i}.gz")
    
filenames["MusicAlbum"] = []
for i in range(5):
    filenames["MusicAlbum"].append(rf"E:\schema_org_datasets\MusicAlbum\MusicAlbum_part_{i}.gz")
    
filenames["QAPage"] = []
for i in range(12):
    filenames["QAPage"].append(rf"E:\schema_org_datasets\QAPage\QAPage_part_{i}.gz")
    
filenames["Recipe"] = []
for i in range(21):
    filenames["Recipe"].append(rf"E:\schema_org_datasets\Recipe\Recipe_part_{i}.gz")
    
filenames["School"] = ["../school_2024.gz"]

filenames["ShoppingCenter"] = ["../shopping_center_2024_p1.gz", "../shopping_center_2024_p2.gz"]

In [ ]:
processed = ["Dataset", "Hospital", "School", "Movie", "ShoppingCenter"]


for cls in schema_org_classes:
    if cls not in processed:
        parse_and_count_min_max_avg_predicates(filenames[cls], cls, "2024", True)
        print()
        parse_and_count_min_max_avg_predicates(filenames[cls], cls, "2024", False)
        
        
        

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 1599
Vidējais predikātu skaits: 1.73

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 20
Vidējais predikātu skaits: 1.66
Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 2629
Vidējais predikātu skaits: 4.91

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 33
Vidējais predikātu skaits: 4.09
Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 1434
Vidējais predikātu skaits: 5.26

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 41
Vidējais predikātu skaits: 4.07
Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 159
Vidējais predikātu skaits: 2.85

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 19
V

**School**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 447
Vidējais predikātu skaits: 4.68

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 20
Vidējais predikātu skaits: 3.87

**ShoppingCenter**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 489
Vidējais predikātu skaits: 6.36

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 18
Vidējais predikātu skaits: 5.6

**Hospital**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 290
Vidējais predikātu skaits: 5.36

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 20
Vidējais predikātu skaits: 4.68

**Dataset**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 4398
Vidējais predikātu skaits: 7.31

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 27
Vidējais predikātu skaits: 4.24

**Movie**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 1813
Vidējais predikātu skaits: 7.74

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 99
Vidējais predikātu skaits: 5.62

**Answer**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 1599
Vidējais predikātu skaits: 1.73

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 20
Vidējais predikātu skaits: 1.66

**Book**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 2629
Vidējais predikātu skaits: 4.91

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 33
Vidējais predikātu skaits: 4.09

**MusicAlbum**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 1434
Vidējais predikātu skaits: 5.26

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 41
Vidējais predikātu skaits: 4.07

**QAPage**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 159
Vidējais predikātu skaits: 2.85

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 19
Vidējais predikātu skaits: 2.73

**Recipe**

Ar atkārtojumiem
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 1752
Vidējais predikātu skaits: 13.21

Bez atkārtojumiem (tikai unikāli predikāti)
Minimālais predikātus skaits: 1
Maksimālais predikātu skaits: 51
Vidējais predikātu skaits: 9.31